src/polyfem/State.cpp

```diff
@@ -966,6 +997,36 @@ namespace polyfem
 		auto it = std::unique(boundary_nodes.begin(), boundary_nodes.end());
 		boundary_nodes.resize(std::distance(boundary_nodes.begin(), it));
 
+		// for elastic pure periodic problem, find an internal node and force zero dirichlet
+		if ((!problem->is_time_dependent() || args["time"]["quasistatic"]) && boundary_nodes.size() == 0 && has_periodic_bc())
```

Comment:
is this the standard way?

Reply:
This is one of the standard ways, another way is to enforce zero mean as a Lagrangian multiplier, which requires solving a saddle point problem.